src/ipfsd-daemon.js

```diff
@@ -246,13 +250,21 @@ class Daemon {
     let output = ''
     this.subprocess = run(this, args, { env: this.env }, {
       error: (err) => {
+        let line = String(err)
+
         // Only look at the last error
-        const input = String(err)
+        const input = line
           .split('\n')
           .map((l) => l.trim())
           .filter(Boolean)
           .slice(-1)[0] || ''
 
+        line = line.trim()
+
+        if (line) {
+          daemonLog.err(line)
```

Comment:
Do you have a particular reason for doing the string trim before? 

I think that the only way it should be used, is if we can receive error strings composed by empty spaces. If this is not the case, what about having the trim only in the log line?

Reply:
The err string has a newline at the end and `debug` will add a newline to anything it's passed so the trim just removes it.  The var isn't used after this so it was just to not do:

```javascript
if (line.trim()) {
   daemonLog.err(line.trim())
}
```

If you prefer I could just do:

```javascript
if (line) {
   daemonLog.err(line.trim())
}
```

Reply:
yes, I would go for your second code snippet.

Reply:
Done.